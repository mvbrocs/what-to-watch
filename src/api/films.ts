import { appAxios } from './';

export interface IFilm {
  name: string;
  poster_image: string;
  preview_image: string;
  background_image: string;
  background_color: string;
  description: string;
  rating: number;
  scores_count: number;
  director: string;
  starring: string[];
  run_time: number;
  genre: string;
  released: number;
  id: number;
  is_favorite: boolean;
  video_link: string;
  preview_video_link: string;
}

export const filmsAPI = {
  getAll: () => appAxios.get<IFilm[]>('/films'),
};
