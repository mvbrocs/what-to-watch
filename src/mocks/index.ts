export { default as mockFilms } from './films';
export { default as mockStore } from './store';
