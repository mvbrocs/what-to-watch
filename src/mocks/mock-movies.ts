import img1 from './img/fantastic-beasts-the-crimes-of-grindelwald.jpg';
import img2 from './img/bohemian-rhapsody.jpg';
import img3 from './img/macbeth.jpg';
import img4 from './img/aviator.jpg';
import img5 from './img/we-need-to-talk-about-kevin.jpg';

export interface IMovie {
  id: string;
  title: string;
  genre: string;
  img: {
    src: string;
    alt: string;
    width: number;
    height: number;
  };
  videoSrc: string;
}

export const mockMovies: IMovie[] = [
  {
    id: 'movie-1',
    title: 'Fantastic Beasts: The Crimes of Grindelwald',
    genre: 'Fantastic',
    img: {
      src: img1,
      alt: 'Fantastic Beasts: The Crimes of Grindelwald',
      width: 280,
      height: 175,
    },
    videoSrc:
      'https://st.depositphotos.com/1100827/3666/v/600/depositphotos_36660843-stock-video-girl-leans-back-among-the.mp4',
  },
  {
    id: 'movie-2',
    title: 'Bohemian Rhapsody',
    genre: 'Dramas',
    img: {
      src: img2,
      alt: 'Bohemian Rhapsody',
      width: 280,
      height: 175,
    },
    videoSrc:
      'https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm',
  },
  {
    id: 'movie-3',
    title: 'Macbeth',
    genre: 'Comedy',
    img: {
      src: img3,
      alt: 'Macbeth',
      width: 280,
      height: 175,
    },
    videoSrc:
      'https://st.depositphotos.com/1100827/3666/v/600/depositphotos_36660843-stock-video-girl-leans-back-among-the.mp4',
  },
  {
    id: 'movie-4',
    title: 'Aviator',
    genre: 'Horror',
    img: {
      src: img4,
      alt: 'Aviator',
      width: 280,
      height: 175,
    },
    videoSrc:
      'https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm',
  },
  {
    id: 'movie-5',
    title: 'What We Do in the Shadows',
    genre: 'Crime',
    img: {
      src: img5,
      alt: 'What We Do in the Shadows',
      width: 280,
      height: 175,
    },
    videoSrc:
      'https://st.depositphotos.com/1100827/3666/v/600/depositphotos_36660843-stock-video-girl-leans-back-among-the.mp4',
  },
];
