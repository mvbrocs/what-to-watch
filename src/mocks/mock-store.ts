import { RootState } from 'state/root-reducer';

export const mockStore: RootState = {
  genre: 'Some genre',
  movies: {
    loaded: false,
    loading: false,
    error: null,
    maxVisible: 0,
    ids: [],
    entities: {},
  },
};
