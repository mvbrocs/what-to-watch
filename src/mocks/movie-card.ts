import { movies } from './movies';

export const movieCardMock = movies[0];
