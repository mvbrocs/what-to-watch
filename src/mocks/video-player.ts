import img1 from './img/fantastic-beasts-the-crimes-of-grindelwald.jpg';

export const videoPlayerAssetsMock = {
  src:
    'https://st.depositphotos.com/1100827/3666/v/600/depositphotos_36660843-stock-video-girl-leans-back-among-the.mp4',
  poster: {
    src: img1,
    alt: 'Fantastic Beasts: The Crimes of Grindelwald',
    width: 280,
    height: 175,
  },
};
