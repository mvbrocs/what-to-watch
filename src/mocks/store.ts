import { RootState } from 'src/state/root-reducer';
import { moviesAdapter, moviesInitialState } from 'src/state/movies';
import { uiInitialState } from 'src/state/ui';
import { userInitialState } from 'src/state/user';
import { history } from 'src/history';

export const mockStore: RootState = {
  router: history,
  ui: uiInitialState,
  movies: moviesAdapter.getInitialState(moviesInitialState),
  user: userInitialState,
};
