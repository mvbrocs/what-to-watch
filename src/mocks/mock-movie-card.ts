import { mockMovies } from './mock-movies';

export const mockMovieCard = mockMovies[0];
