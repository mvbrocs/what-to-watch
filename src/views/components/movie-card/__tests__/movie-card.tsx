import React from 'react';
import { shallow } from 'enzyme';

import { MovieCard } from '../index';
import { mockFilms } from 'mocks/films';

describe('MovieCard', () => {
  const props = {
    name: mockFilms[0].name,
    preview_image: mockFilms[0].preview_image,
    preview_video_link: mockFilms[0].preview_video_link,
  };
  const movieCard = shallow(<MovieCard {...props} />);

  it('renders properly', () => {
    expect(movieCard).toMatchSnapshot();
  });

  it('should render VideoPlayer with play=true prop, on mouseEnter', () => {
    movieCard.simulate('mouseEnter');

    expect(movieCard.find('VideoPlayer').prop('play')).toBeTruthy();
  });

  it('should render VideoPlayer with play=false prop, on mouseLeave', () => {
    movieCard.simulate('mouseLeave');

    expect(movieCard.find('VideoPlayer').prop('play')).toBeFalsy();
  });
});
