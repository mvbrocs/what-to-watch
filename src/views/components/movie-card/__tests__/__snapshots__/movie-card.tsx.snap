// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders properly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <VideoPlayer
      delay={1000}
      play={false}
      poster={
        Object {
          "alt": "Orlando",
          "height": 175,
          "src": "https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/orlando.jpg",
          "width": 280,
        }
      }
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <Link
      className="small-movie-card__link"
      to="/films/1"
    >
      Orlando
    </Link>
  </h3>
</article>
`;
